### PR TITLE
M2286 Settings management for plugins and app

### LINKED ISSUE #3454: Genome browser highlight region option broken
Reproduce:
- Enable highlight region option for genome browser
- Select genomic data set in genome browser

Expected:
- Pinkish overlay on genome browser region

Actual:
- No changes

### LINKED ISSUE #3455: Updating EntityMetaData bean does not update (meta)data repo
Reproduce:
- Add a nillable string attribute to ScriptMeta

Expected:
- After app restart the attributes table is updated and the Script table had a new column with null values

Actual:
- Nothing happens

### LINKED ISSUE #3456: Google Analytics tracking incorrect in case of two tracking snippets
Reproduce:
- Include Google Analytics snippets for two accounts (e.g. MOLGENIS account and customer account)

Expected:
- Tracking takes place via one tracking snippet

Actual:
- Tracking takes place through two tracking snippets, according to Google Analytics documentation tracking should take place via one snippet and different tracker names.

### PR DESCRIPTION
**Pull request by dennishendriksen**
- Fixed #3454 Genome browser highlight region option broken
- Fixed #3455 Updating EntityMetaData bean does not update (meta)data repo
- Fixed #3456 Google Analytics tracking incorrect in case of two tracking snippets

### CODE REVIEW

File: molgenis-app/src/main/java/org/molgenis/app/controller/BackgroundController.java [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 import static org.molgenis.app.controller.BackgroundController.URI;
 
-import org.molgenis.framework.ui.MolgenisPluginController;
+import org.molgenis.ui.MolgenisPluginController;
```
Comment (tommydeboer) [inline molgenis-app/src/main/java/org/molgenis/app/controller/BackgroundController.java]: Just curious, why rename?

File: molgenis-security/src/main/java/org/molgenis/security/account/AccountController.java [hunk 2]
```diff
@@ -85,7 +86,7 @@ public ModelAndView getRegisterForm()
 	{
 		ModelAndView model = new ModelAndView("register-modal");
 		model.addObject("countries", CountryCodes.get());
-		model.addObject("min_password_length", MIN_PASSWORD_LENGTH);
+		model.addObject("min_password_length", 6); // FIXME do not hardcode
```
Comment (tommydeboer) [inline molgenis-security/src/main/java/org/molgenis/security/account/AccountController.java]: Leave this FIXME for now?

File: molgenis-security/src/main/java/org/molgenis/security/account/RegisterRequest.java [hunk 3]
```diff
@@ -12,10 +10,10 @@
 	@NotNull
 	private String username;
 	@NotNull
-	@Size(min = MIN_PASSWORD_LENGTH)
+	@Size(min = 6) // FIXME do not hardcode
 	private String password;
 	@NotNull
-	@Size(min = MIN_PASSWORD_LENGTH)
+	@Size(min = 6) // FIXME do not hardcode
```
Comment (tommydeboer) [inline molgenis-security/src/main/java/org/molgenis/security/account/RegisterRequest.java]: Leave FIXME for now?

File: molgenis-data-migrate/src/main/java/org/molgenis/data/migrate/v1_9/RuntimePropertyToAppSettingsMigrator.java [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+				String key = "app.trackingcode.header";
+				RuntimeProperty property = getProperty(key);
+				if (property != null)
+				{
+					String rtpValue = property.getValue();
+					String value = appSettings.getTrackingCodeFooter();
+					if (!rtpValue.equals(value))
+					{
+						LOG.info("Updating AppSettings for RuntimeProperty [" + key + "]");
+						appSettings.setTrackingCodeFooter(value);
```
Comment (tommydeboer) [inline molgenis-data-migrate/src/main/java/org/molgenis/data/migrate/v1_9/RuntimePropertyToAppSettingsMigrator.java]: Header codes get migrated to the footer...

File: molgenis-data-migrate/src/main/java/org/molgenis/data/migrate/v1_9/RuntimePropertyToAppSettingsMigrator.java [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+				String key = "app.trackingcode.footer";
+				RuntimeProperty property = getProperty(key);
+				if (property != null)
+				{
+					String rtpValue = property.getValue();
+					String value = appSettings.getTrackingCodeFooter();
+					if (!rtpValue.equals(value))
+					{
+						LOG.info("Updating AppSettings for RuntimeProperty [" + key + "]");
+						appSettings.setGoogleAnalyticsTrackingId(value);
```
Comment (tommydeboer) [inline molgenis-data-migrate/src/main/java/org/molgenis/data/migrate/v1_9/RuntimePropertyToAppSettingsMigrator.java]: ...and overwritten by trackingcode.footer

**Comment by tommydeboer** [comment]
Misses a migration step for AttributeMapping (*algorithm* attribute's dataType was changed to TEXT). Will close this pull request and make a new one from my branch.

**Comment by dennishendriksen** [comment]
Missing migration step is separate issue, see #3417.

**Pull request merged**